pkg/mdformatter/linktransformer/validator.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Bartłomiej Płotka @bwplotka
+// Licensed under the Apache License 2.0.
+
+package linktransformer
+
+import (
+	"strconv"
+	"strings"
+)
+
+type Validator interface {
+	IsValid(URL string) (bool, error)
+}
+
+// GitHubValidator.IsValid skips visiting all github issue/PR links.
+func (v GitHubValidator) IsValid(URL string) (bool, error) {
+	// Find rightmost index of match i.e, where regex match ends.
+	// This will be where issue/PR number starts. Split incase of section link and convert to int.
+	rightmostIndex := v._regex.FindStringIndex(URL)
+	stringNum := strings.Split(URL[rightmostIndex[1]:], "#")
+	num, err := strconv.Atoi(stringNum[0])
+	if err != nil {
+		return false, err
+	}
+	// If number in link does not exceed then link is valid.
+	if v._maxNum >= num {
+		return true, nil
+	}
+	return false, nil
+}
+
+// RoundTripValidator.IsValid returns false if url matches, to ensure it is visited by colly.
+func (v RoundTripValidator) IsValid(URL string) (bool, error) {
```

Comment:
This still needs some love 🙈 

"RountTripValidator says URL is invalid" - that does not sounds like this would mean it will be check by colly. I would rather put colly logic directly here into this method, no?

Reply:
Alright, will put this logic directly! 🙂